### DISCUSSION TITLE
How to force `https` request only on gateway?

### DISCUSSION THREAD

**Discussion by mchenx** [ThreeMammals/Ocelot#1922] [Q&A] [answered]
## Expected Behavior / New Feature
There is RequireHttpsAttribute in ASP.Net MVC, am wondering if we can have UpStreamScheme  that can be defined as https in reroute, so that when the client side sends request with http instead of https protocol, the gateway can refuse it. 

I just want to get rid of the RequireHttpsAttribute on the web api service that is running behind the gateway, I don't want to configure certifications on that service, just want to use http calls from gateway to that backedn service, and limit https requests only in the front, on gateway.

Am I missing something here? 
Kindly advise..

## Actual Behavior / Motivation for New Feature
Currently there seems no way to define a UpstreamPath that has to be https only.

**Comment by raman-m** [comment, marked as answer]
Hi @mchenx !
Welcome to Ocelot world! :tiger2: 

You can develop custom [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) to replace `https` scheme with `http` one. But in route you will have `https`.
